### PR TITLE
Fix(engine): Return revert error message during contract deploy

### PR DESCRIPTION
**Pull request by birchmd**
@0x3bfc pointed out that presently the Engine always returns the address the contract would be deployed at, regardless of if the deploy execution was successful or not. In this PR we ensure to pass the error message on in the case of failure.

### CODE REVIEW

File: engine-precompiles/Cargo.toml [hunk 1]
```diff
@@ -19,8 +19,8 @@ base64 = { version = "0.13.0", default-features = false, features = ["alloc"] }
 blake2 = { git = "https://github.com/near/near-blake2.git", version = "0.9.1", default-features = false }
 borsh = { version = "0.8.2", default-features = false }
 bn = { package = "aurora-bn", git = "https://github.com/aurora-is-near/aurora-bn.git", default-features = false }
-evm = { git = "https://github.com/aurora-is-near/sputnikvm.git", default-features = false }
-evm-core = { git = "https://github.com/aurora-is-near/sputnikvm.git", default-features = false }
+evm = { git = "https://github.com/aurora-is-near/sputnikvm.git", rev = "177e253ad947b7186bfa6920843075556b60ff54", default-features = false }
```
Comment (birchmd) [inline engine-precompiles/Cargo.toml:22]: Switching to use our EVM fork for now, we can return to the upstream master branch once https://github.com/rust-blockchain/evm/pull/91 is merged.

File: engine-precompiles/src/lib.rs [hunk 2]
```diff
@@ -47,9 +47,9 @@ impl PrecompileOutput {
     }
 }
 
-impl From<PrecompileOutput> for evm::executor::PrecompileOutput {
+impl From<PrecompileOutput> for executor::stack::PrecompileOutput {
```
Comment (birchmd) [inline engine-precompiles/src/lib.rs:50]: These changes are unrelated to the core goal of this PR to properly handle reverts during contract deploy, but were needed after getting up to date with the latest upstream changes of SputnikVM (and getting up to date there was needed for this fix because I needed to make a small change to SputnikVM for this to work properly).

**Review by joshuajbouw — APPROVED** [review]

**Review by 0x3bfc — APPROVED** [review]
LGTM .. Thanks @birchmd 👍
